dora/core/server/master/src/main/java/alluxio/master/scheduler/Scheduler.java

```diff
@@ -414,38 +514,32 @@ public WorkerInfoHub(Scheduler scheduler, WorkerProvider workerProvider) {
       mWorkerProvider = workerProvider;
     }
 
-    private final Map<WorkerInfo, PriorityBlockingQueue<Task>> mWorkerToTaskQ
+    private final Map<WorkerInfoIdentity, BoundedPriorityBlockingQueue<Task>> mWorkerToTaskQ
         = new ConcurrentHashMap<>();
 
     /**
-     * Enqueue task for worker.
-     * @param workerInfo the worker
-     * @param task the task
-     * @param kickStartTask kick-start task
-     * @return whether the task is enqueued successfully
+     * Kick stark tasks for each worker task q.
      */
-    public boolean enqueueTaskForWorker(@Nullable WorkerInfo workerInfo, Task task,
-        boolean kickStartTask) {
-      if (workerInfo == null) {
-        return false;
-      }
-      PriorityBlockingQueue workerTaskQ = mWorkerToTaskQ
-          .computeIfAbsent(workerInfo, k -> new PriorityBlockingQueue<>());
-      if (!workerTaskQ.offer(task)) {
-        return false;
-      }
-      CloseableResource<BlockWorkerClient> blkWorkerClientResource = mActiveWorkers.get(workerInfo);
-      if (blkWorkerClientResource == null) {
-        LOG.warn("Didn't find corresponding BlockWorkerClient for workerInfo:{}",
-            workerInfo);
-        return false;
-      }
-      if (kickStartTask) {
-        // track running tasks of a job
-        ConcurrentHashSet<Task<?>> tasks = mJobToRunningTasks.computeIfAbsent(task.getJob(),
-                j -> new ConcurrentHashSet<>());
-        tasks.add(task);
-        task.execute(blkWorkerClientResource.get(), workerInfo);
+    public void kickStartTasks() {
+      // Kick off one task for each worker
+      mWorkerToTaskQ.forEach((workerInfo, tasksQ) -> {
+        LOG.debug("Kick start task for worker:{}, taskQ size:{}",
+            workerInfo.mWorkerInfo.getAddress().getHost(),
+            tasksQ.size());
+        CloseableResource<BlockWorkerClient> blkWorkerClientResource
+            = mActiveWorkers.get(workerInfo);
+        if (blkWorkerClientResource == null) {
+          LOG.debug("Didn't find corresponding BlockWorkerClient for workerInfo:{}",
```

Comment:
any reason we change it to debug? this seems to be a serious problem if client is not there

Reply:
changed back to warn